packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -220,7 +222,14 @@ Future<int> startApp(
   }
 
   if (!startedSomething) {
-    printError('Unable to run application. Please ensure that a supported device/simulator is connected.');
+    int connected = devices.all.where((device) => device.isConnected()).length;
+    String message = 'Unable to run application';
+    if (connected == 0) {
+      message += ' - no connected devices.';
+    } else if (unsupportedCount != 0) {
+      message += ' - $unsupportedCount unsupported device(s) connected';
```

Comment:
In `commands/devices.dart` there is a `pluralize()` method you can use to print out either `device` or `devices` depending on the count.